### PR TITLE
Use flash element from theme if available.

### PR DESCRIPTION
**Pull request by ADmad**
Refs #5920

### CODE REVIEW

**Comment by nrother** [comment]
Maybe you forgot to commit the flash elements in the TestTheme?

**Comment by ADmad** [comment]
@nrother Yup i did :P

File: tests/TestCase/View/Helper/FlashHelperTest.php [hunk 1]
```diff
@@ -154,4 +154,19 @@ public function testFlashWithPluginElement()
...
+     * @return void
+     */
+    public function testFlashWithTheme()
+    {
+        Plugin::load('TestTheme');
+
+        $this->View->theme = 'TestTheme';
+        $result = $this->Flash->render('flash');
+        $expected = 'flash element from TestTheme';
+        $this->assertContains($expected, $result);
```
Comment (AD7six) [inline tests/TestCase/View/Helper/FlashHelperTest.php]: The changes in this PR don't do anything - This test will pass using 3.0's flash helper.

```
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 $=)
-> git status
# On branch issue-5920
nothing to commit (working directory clean)
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 $=)
-> phpunit tests/TestCase/View/Helper/FlashHelperTest.php 
PHPUnit 4.3.4 by Sebastian Bergmann.

Configuration read from /home/andy/repos/cakephp/cakephp/phpunit.xml.dist

.....

Time: 106 ms, Memory: 6.75Mb

OK (5 tests, 8 assertions)
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 $=)
-> git checkout origin/3.0 src/
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 +$=)
-> git diff --cached
diff --git src/View/Helper/FlashHelper.php src/View/Helper/FlashHelper.php
index 7fd83ef..f871c94 100644
--- src/View/Helper/FlashHelper.php
+++ src/View/Helper/FlashHelper.php
@@ -80,11 +80,7 @@ class FlashHelper extends Helper
         $flash = $options + $flash;
         $this->request->session()->delete("Flash.$key");
 
-        $element = $flash['element'];
-        if ($this->_View->theme && strpos($element, '.') === false) {
-            $element = $this->_View->theme . '.' . $element;
-        }
-        return $this->_View->element($element, $flash);
+        return $this->_View->element($flash['element'], $flash);
     }
 
     /**
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 +$=)
-> phpunit tests/TestCase/View/Helper/FlashHelperTest.php 
PHPUnit 4.3.4 by Sebastian Bergmann.

Configuration read from /home/andy/repos/cakephp/cakephp/phpunit.xml.dist

.....

Time: 109 ms, Memory: 7.00Mb

OK (5 tests, 8 assertions)
www-data @ dev [ /home/andy/repos/cakephp/cakephp ] (issue-5920 +$=)
-> 
```

**Comment by AD7six** [comment]
The view theme applies to all rendered templates, unless I'm missing something this doesn't actually change/do anything: :-1:

**Comment by ADmad** [comment]
Heh, on the bright side at least we now have a test showing no change is needed :)  I'll amend the PR.

**Comment by nrother** [comment]
Ok, I just checked with my setup, and it really works without any changes. Sorry for the confusion.
(Actually it did not work on my setup because I never set the `$theme` and directly searched for the config option in the flash system. Definitely something for the docs.)

**Comment by AD7six** [comment]
Moar test coverage, moar awesome :+1:

**Pull request merged**